src/openrct2/core/DataSerialiserTraits.h

```diff
@@ -730,3 +731,46 @@ template<> struct DataSerializerTraits_t<rct_vehicle_colour>
         stream->Write(msg, strlen(msg));
     }
 };
+
+template<> struct DataSerializerTraits_t<ObjectEntryDescriptor>
+{
+    static void encode(OpenRCT2::IStream* stream, const ObjectEntryDescriptor& val)
+    {
+        stream->Write(&val.Generation);
+        if (val.Generation == ObjectGeneration::DAT)
+        {
+            DataSerializerTraits<rct_object_entry> s;
+            s.encode(stream, val.Entry);
+        }
+        else
+        {
+            DataSerializerTraits<std::string> s;
+            s.encode(stream, val.Identifier);
+        }
+    }
+    static void decode(OpenRCT2::IStream* stream, ObjectEntryDescriptor& val)
+    {
+        ObjectGeneration generation;
+        stream->Read(&generation);
+        if (generation == ObjectGeneration::DAT)
+        {
+            rct_object_entry obj;
+            DataSerializerTraits<rct_object_entry> s;
+            s.decode(stream, obj);
+            val = ObjectEntryDescriptor(obj);
+        }
+        else
+        {
+            std::string id;
+            DataSerializerTraits<std::string> s;
+            s.decode(stream, id);
+            val = ObjectEntryDescriptor(id);
+        }
+    }
+    static void log(OpenRCT2::IStream* stream, const ObjectEntryDescriptor& val)
+    {
+        char msg[128] = {};
+        snprintf(msg, sizeof(msg), "ObjectEntryDescriptor (Generation = %d)", static_cast<int32_t>(val.Generation));
```

Comment:
This won't log the identifier/DAT name, which is the most interesting bit.

This _might_ be fine, I don't know how important this logging function is.

Reply:
Loging function is only used in network code as far as i am aware so you can't actually trigger this.